src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -232,45 +240,49 @@ public async Task<PrStatus> GetPullRequestStatusAsync(string pullRequestUrl)
                 {
                     return PrStatus.Closed;
                 }
-            }
 
-            return PrStatus.None;
-        }
-
-        public async Task<string> GetPullRequestRepo(string pullRequestUrl)
-        {
-            string url = GetPrPartialAbsolutePath(pullRequestUrl);
-
-            HttpResponseMessage response = await this.ExecuteGitCommand(HttpMethod.Get, url, _logger);
-
-            JObject responseContent = JObject.Parse(await response.Content.ReadAsStringAsync());
+                throw new DarcException($"Unhandled Azure DevOPs PR status {status}");
+            }
 
-            return responseContent["repository"]["remoteUrl"].ToString();
+            throw new DarcException($"Failed to parse PR status: {responseContent["status"]}");
         }
 
+        /// <summary>
+        ///     Retrieve information on a specific pull request
+        /// </summary>
+        /// <param name="pullRequestUrl">Uri of the pull request</param>
+        /// <returns>Information on the pull request.</returns>
         public async Task<PullRequest> GetPullRequestAsync(string pullRequestUrl)
         {
-            VssConnection connection = CreateConnection(pullRequestUrl);
-            GitHttpClient client = await connection.GetClientAsync<GitHttpClient>();
+            (string accountName, string projectName, string repoName, int id) = ParsePullRequestUri(pullRequestUrl);
 
-            (string team, string repo, int id) = ParsePullRequestUri(pullRequestUrl);
+            VssConnection connection = CreateVssConnection(accountName);
+            GitHttpClient client = await connection.GetClientAsync<GitHttpClient>();
 
-            GitPullRequest pr = await client.GetPullRequestAsync(team, repo, id);
+            GitPullRequest pr = await client.GetPullRequestAsync(projectName, repoName, id);
             return new PullRequest
             {
                 Title = pr.Title,
                 Description = pr.Description,
-                BaseBranch = pr.TargetRefName,
-                HeadBranch = pr.SourceRefName
+                // Strip out the refs/heads prefix on BaseBranch and HeadBranch because almost
+                // all of the other APIs we use do not support them (e.g. get an item at branch X).
+                BaseBranch = pr.TargetRefName.Replace("refs/heads/", ""),
```

Comment:
This looks like another case of "Contains" in disguise. Technically I could be a jerk and name my branch "refs/heads/pizza", and this replace would mangle my branch names.

People won't do that, but it's good to be diligent about doing what we mean to, because someone might copy/paste this code somewhere else where it does matter.

Reply:
Yeah it should be changed to just replace the first instance.  I thought refs/heads was a reserved prefix in git.  But yes you can be a jerk and name your branch like that.